cli/internal/cloudcmd/create.go

```diff
@@ -215,14 +217,42 @@ func (c *Creator) createAzure(ctx context.Context, cl terraformClient, config *c
 		return clusterid.File{}, err
 	}
 
+	// very, very hacky way to update the MAA attestation policy. This should be changed as soon as either the Terraform provider supports it
+	// or the Go SDK gets updated to a recent API version.
+	// https://github.com/hashicorp/terraform-provider-azurerm/issues/20804
+	cmd := exec.CommandContext(ctx,
+		"az", "rest",
```

Comment:
Having runtime dependencies like this is too hacky for main imo.

If we want to merge nevertheless, this is missing appropriate documentation about az as required dependency.

Reply:
If I'm not mistaken the [Azure CLI should be a requirement](https://docs.edgeless.systems/constellation/getting-started/install#authentication) already?

Reply:
For authentication, yes. For running our CLI, no. 
There are other ways to authenticate against Azure, like the Azure login action, that don't require to have az available. We should explicitly state that this required to execute the CLI (which is also a breaking change, btw).

Reply:
Let's talk about that tomorrow. I may not be as well informed about which parts of the setup process need to have this available and which do not, but I think for executing the Terraform file (which also happens at cluster creation time), the Azure CLI is required? Nonetheless docs are necessary though, I agree

Reply:
If I may give my 2.5 cents to this:
>There are other ways to authenticate against Azure, like the Azure login action, that don't require to have az available.
I think the Azure login action also requires the CLI exist.
However, Terraform can directly authenticate over a Service Principal and Client Secret, what we currently use for e2e tests for example. If I understand correctly, _that_ doesn't need the Azure CLI to be present (in PATH).

However, I might have a suggestion for a slightly less hacky version:
Use `azidentity` from the Azure Go SDK to authenticate with Azure, retrieve the Bearer token from it (that is directly exportable) and just manually do a PUT request using Go's http library. Then you can use the SDK to authenticate and don't need the Azure CLI necessarily if you fall back to other authentication methods the SDK supports.

However, then you still need to keep in mind that Terraform expects the credentials to be prefixed with `ARM` while `azidentity` doesn't. But some glue code might be able to copy from one to another if they can be used equally otherwise.

Reply:
> Or maybe the Azure SDK also offers a Go HTTP client that is authenticated
That's an idea I also came across, the [`autorest`](https://pkg.go.dev/github.com/Azure/go-autorest/autorest#Prepare) package does exactly that.
The only thing I'm concerned about is: How would we retrieve the login info from the user? Doesn't he also need the CLI installed for that?

Reply:
What was the outcome of Azure/azure-sdk-for-go#20349 ?

Reply:
Found that the [`attestation`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/attestation/2020-10-01/attestation) package should provide this functionality, but it's outdated as well (doesn't support auth adapters nor AzureVM TEE type, while latter can be hacked around though)

Reply:
If they would regenerate it, could we use it instead of this hack? If yes, maybe you should reopen it?

Reply:
I do not think so. 

There are multiple layers to the problem:

1. **Outdated Azure REST API** 
	The documentation of the [Azure REST API](https://learn.microsoft.com/en-us/rest/api/attestation/policy/set?tabs=HTTP) (from which I believe the SDKs get generated?) does not mention that Azure VMs ( `attestationType=AzureGuest` ) are supported. I only found the type string "by accident" when inspecting the HTTP request of the Azure web interface. Furthermore, the Azure docs do not mention the Azure VMs anywhere, as far as I found out. The API docs claim to be on API version `2022-08-01`.
2. **Outdated Go SDKs**
	Both the [`armattestation`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/attestation/armattestation) as well as the [`attestation`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/attestation/2020-10-01/attestation) package provide an API to interact with Azure attestation providers. Both packages have the problem of not explicitly supporting the Azure VM type. This can be hacked around though by just inserting the string `AzureGuest` since in both packages, the `TeeType` implementation boils down to being just a string internally. For the `armattestation` package, the next problem is that the package does not support policy management at all. The only thing an [update request](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/attestation/armattestation#ServicePatchParams) can contain is a map of tags. For the `attestation` package, there is another problem. First, it does not support credential auth (like any other Azure SDK), but only the old [`adal`](https://pkg.go.dev/github.com/Azure/go-autorest/autorest#NewBearerAuthorizer) OAuth, which will be EOL on March 31st. I thought this can be hacked around by creating a shim which implements the interface, but it turned out that the [`WithAuthorization`](https://github.com/Azure/azure-sdk-for-go/blob/2187aca40bbf6931ffac666977e5d29128d82585/services/attestation/2020-10-01/attestation/policy.go#L252) method is just not being called at all. Both packages claim to be on API version `2020-10-01`.
3. **Outdated Terraform provider** 
	The functionality provided by the [Azure Terraform provider](https://registry.terraform.io/providers/hashicorp/azurerm/3.46.0/docs/resources/attestation) is similar to what the `armattestation` package can do. There's just not support for policy management at all. I don't know how the Terraform provider gets generated internally, but I assume it would also use either the REST API definitions or the SDKs? However, the functionality is not sufficient as is.

I don't know how I would proceed with this. I assume a request to first-level-support asking for the API to be updated. Once that is in we could try and get the SDKs updated. Either way, this process will surely not be fast.. 

Let me know what you think @thomasten @azure-proficient-folks 😄

Reply:
According to Daniel, for cluster creation the Azure CLI currently is not needed. I don't know how Terraform authenticates with Azure exactly, but maybe we could require to set credentials when executing cluster creation with MAA selected as attestation type and then authenticate with the Azure API and send the HTTP request as @Nirusu said?

Reply:
I *think* the Azure CLI is not needed. Terraform should be able to load credentials from file or environment. The azure CLI would only be "needed" to generate those credentials in the first place.
The Azure SDK has a credentials loading function that handles looking for credentials in all the common places (this also does not require the Azure CLI), I would assume Terraform uses this function (or something similar).

Using those credentials to send HTTP request might work

Reply:
Just verified: Constellation CLI is needed for cluster creation.

```
moritz@moritzs-qemu-test:~/const/const-setup$ constellation create --control-plane-nodes 1 --worker-nodes 2 -y
Creating
An error occurred: exit status 1

Error: building AzureRM Client: please ensure you have installed Azure CLI version 2.0.79 or newer. Error parsing json result from the Azure CLI: launching Azure CLI: exec: "az": executable file not found in $PATH.

Reply:
It is needed if you don't specify any environment variables with credentials (see what we do in the e2e-tests) or used managed identities. But yes, the _default_ authentication method is over the CLI.